Specify WHOIS and all its known numerics (#115)
ircdocs/modern-irc: index.md

One thing for bot authors to note is that the `NOTICE` message may be interpreted differently by various clients. Some clients highlight or interpret any `NOTICE` sent to a channel in the same way that a `PRIVMSG` with their nickname gets interpreted. This means that users may be irritated by the use of `NOTICE` messages rather than `PRIVMSG` messages by clients or bots, and they are not commonly used by client bots for this reason.

## User-Based Queries

### WHOIS message

         Command: WHOIS
      Parameters: [<target>] <nick>

This command is used to query information about particular users.
The server will answer this command with several numeric messages with information about the nicks, ending with [`RPL_ENDOFWHOIS`](#rplendofwhois-318).

Servers MUST end their reply to `WHOIS` messages with one of these numerics:

* [`ERR_NOSUCHNICK`](#errnosuchnick-401) `(401)`
* [`ERR_NOSUCHSERVER`](#errnosuchserver-402) `(402)`
* [`ERR_NONICKNAMEGIVEN`](#errnonicknamegiven-431) `(431)`
* [`RPL_ENDOFWHOIS`](#rplendofwhois-318) otherwise, even if they did not send any other numeric message. This allows clients to stop waiting for new numerics.

> Client MUST NOT not assume all numerics messages are sent at once, so server can interleave other messages before the end of the WHOIS response.

Reviewer: ```suggestion
Client MUST NOT not assume all numeric messages are sent at once, as server can interleave other messages before the end of the WHOIS response.
```

If the `<target>` parameter is specified, it SHOULD be a server name or the nick of a user. Servers SHOULD send the query to a specific server with that name, or to the server `<target>` is connected to, respectively.

> 
> Typically, it is useful to client who want to know how long the user in question has been idle as only local server (i.e., the server the user is directly connected to) knows that information, while everything else is globally known.

Reviewer: ```suggestion
Typically, it is used by clients who want to know how long the user in question has been idle (as typically only the server the user is directly connected to knows that information, while everything else this command returns is globally known).
```
Makes sense to push this on the end of the `<target>`  paragraph, since this is just clarifying why that would be used.

The following numerics MAY be returned as part of the whois reply:

* [`RPL_WHOISCERTFP`](#rplwhoiscertfp-276) `(276)`
* [`RPL_WHOISREGNICK`](#rplwhoisregnick-307) `(307)`
* [`RPL_WHOISUSER`](#rplwhoisuser-311) `(311)`
* [`RPL_WHOISSERVER`](#rplwhoisserver-312) `(312)`
* [`RPL_WHOISOPERATOR`](#rplwhoisoperator-313) `(313)`
* [`RPL_WHOISIDLE`](#rplwhoisidle-317) `(317)`
* [`RPL_WHOISCHANNELS`](#rplwhoischannels-319) `(319)`
* [`RPL_WHOISSPECIAL`](#rplwhoisspecial-320) `(320)`
* [`RPL_WHOISACCOUNT`](#rplwhoisaccount-330) `(330)`

> * [`RPL_WHOISACTUALLY`](#rplwhoisactually-338) `(338)`

Reviewer: Hmm, might be useful to mention that a couple of these (`RPL_WHOISACTUALLY`, maybe CERTFP?, etc) can typically only be seen by server operators.

One thing for bot authors to note is that the `NOTICE` message may be interpreted differently by various clients. Some clients highlight or interpret any `NOTICE` sent to a channel in the same way that a `PRIVMSG` with their nickname gets interpreted. This means that users may be irritated by the use of `NOTICE` messages rather than `PRIVMSG` messages by clients or bots, and they are not commonly used by client bots for this reason.

## User-Based Queries

### WHOIS message

         Command: WHOIS
      Parameters: [<target>] <nick>

This command is used to query information about particular users.
The server will answer this command with several numeric messages with information about the nicks, ending with [`RPL_ENDOFWHOIS`](#rplendofwhois-318).

Servers MUST end their reply to `WHOIS` messages with one of these numerics:

* [`ERR_NOSUCHNICK`](#errnosuchnick-401) `(401)`
* [`ERR_NOSUCHSERVER`](#errnosuchserver-402) `(402)`
* [`ERR_NONICKNAMEGIVEN`](#errnonicknamegiven-431) `(431)`

> * [`RPL_ENDOFWHOIS`](#rplendofwhois-318) otherwise, even if they did not send any other numeric message. This allows clients to stop waiting for new numerics.

Reviewer 2: Maybe add the numeric number in parentheses, just like the rest of the list items?
